client/src/main/java/au/org/consumerdatastandards/client/api/ProtectedAPI.java

```diff
@@ -0,0 +1,53 @@
+package au.org.consumerdatastandards.client.api;
+
+import au.org.consumerdatastandards.client.ApiClient;
+
+
+import java.net.DatagramSocket;
+import java.net.InetAddress;
+import java.net.SocketException;
+import java.net.UnknownHostException;
+import java.time.OffsetDateTime;
+import java.util.Base64;
+import java.util.Map;
+
+public class ProtectedAPI {
+
+    protected ApiClient apiClient;
+
+    public ProtectedAPI() {
+        this(new ApiClient());
+    }
+
+    public ProtectedAPI(ApiClient apiClient) {
+        this.apiClient = apiClient;
+    }
+
+    protected void addCdsProtectedApiHeaders(Map<String, String> headers) {
+        headers.put("x-fapi-customer-ip-address", getIpAddress());
+        headers.put("x-fapi-auth-date", OffsetDateTime.now().toString());
+        headers.put("x-cds-user-agent", Base64.getEncoder().encodeToString(apiClient.getUserAgent().getBytes()));
+        headers.put("x-cds-subject", "cds-client-cli");
```

Comment:
I think it's better not to include this header. It's doomed: https://github.com/ConsumerDataStandardsAustralia/standards-maintenance/issues/33 and it's supposed to be the same as the _sub_ claim

Reply:
yes, I am aware of that. This is for 1.0.0 release. We'll change that in 1.0.1 release. I include it because data-holder validates it as well. It won't break things even if banks don't required it.

Reply:
Well, then you should get the sub value from the access token.